src/main/java/com/example/boot/service/impl/ConverterImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.example.boot.service.impl;
+
+import com.example.boot.model.Currency;
+import com.example.boot.service.Converter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConverterImpl implements Converter {
+    @Value("${service.url}")
+    private String url;
+
+    @Override
+    public BigDecimal convert(Currency from, Currency to, BigDecimal amount) {
+        url = url + "?from=" + from + "&to=" + to + "&amount=" + amount;
```

Comment:
let's better not change the field - but create a local variable here

and use String.format()